R/render.R

```diff
@@ -367,18 +367,34 @@ render <- function(input,
 
       params <- knit_params_get(input_lines, params)
 
+      # bail if an object called 'params' exists in this environment,
+      # and it seems to be an unrelated user-created object. store
+      # references so we can restore them post-render
+      hasParams <- exists("params", envir = envir, inherits = FALSE)
+      envirParams <- NULL
+
+      if (hasParams) {
+        envirParams <- get("params", envir = envir, inherits = FALSE)
+        isKnownParamsObject <-
+          inherits(envirParams, "knit_param_list") ||
+          inherits(envirParams, "knit_param")
```

Comment:
FYI the second argument of `inherits()` can be a character vector, so you can get rid of `||` if you want.